### PR TITLE
Fit DenseNet examples to Keras-2

### PR DESCRIPTION
**Pull request by kemaswill**

### CODE REVIEW

**Comment by ahundt** [comment]
https://github.com/fchollet/keras/pull/6034 may be relevant. Also a much more complete version of these changes is in https://github.com/farizrahman4u/keras-contrib/pull/46/files#diff-7c92da79e0a644cc9e206dab6de13bab

**Comment by farizrahman4u** [comment]
LGTM, thanks!

File: keras_contrib/applications/densenet.py [hunk 1]
```diff
@@ -375,15 +376,16 @@ def __dense_block(x, nb_layers, nb_filter, growth_rate, bottleneck=False, dropou
         x = __conv_block(x, growth_rate, bottleneck, dropout_rate, weight_decay)
         x_list.append(x)
 
-        x = merge(x_list, mode='concat', concat_axis=concat_axis)
+        x1 = concatenate(x_list, axis=concat_axis)
```
Comment (ahundt) [inline keras_contrib/applications/densenet.py]: Not good! then all the convolutions will be the same! it is essential that all the concatenation be on x.

Comment (ahundt) [inline keras_contrib/applications/densenet.py]: Specifically, this is the source of fchollet/keras#6034

Comment (kemaswill) [inline keras_contrib/applications/densenet.py]: Hmm, interesting. So I should wait for this PR to be merged, right?

Comment (ahundt) [inline keras_contrib/applications/densenet.py]: Unfortunately, this change both triggers an existing bug and adds a new one. Consider just utilizing/merging #46 it fixes everything mentioned here, and I'll be switching to concat calls once fchollet/keras#6034 is merged.

Comment (kemaswill) [inline keras_contrib/applications/densenet.py]: Sure, I will take a look. Thanks.

**Comment by kemaswill** [comment]
@ahundt Has you code be merged? If yes, then we don't need the revert any more, right?

**Comment by ahundt** [comment]
@kemaswill I'm still waiting on review for #46

**Comment by kemaswill** [comment]
Got it.

**Pull request merged**